patients/repo.go

```diff
@@ -1289,11 +1289,17 @@ func (r *repository) TideReport(ctx context.Context, clinicId string, params Tid
 	}
 
 	{
+		// This specifically catches users who:
+		// 1. Have no data within the last 8h
+		// AND either of the following:
+		// 2. Have no data within the cutoff, typically the period length being looked at, subtracted from now
+		// 3. Have a dexcom session, and it is not successfully connected
 		selector := bson.M{
-			"clinicId": clinicObjId,
-			"tags":     bson.M{"$all": tags},
+			"clinicId":                        clinicObjId,
+			"tags":                            bson.M{"$all": tags},
+			"summary.cgmStats.dates.lastData": bson.M{"$lt": time.Now().UTC().Add(-8 * time.Hour)},
```

Comment:
what if LastDataCutoff is say... -10 hours and the user's last data cgmStats is -9 hours (and their dexcom conn is fine)

They wouldn't match this filter... is that the intended behavior? Line 1300 makes me think it's not. Should there be a min((now -8 hours), params.LastDataCutoff)?